bloomstack_core/hook_events/stock_entry.py

```diff
@@ -17,3 +17,30 @@ def add_comment_to_batch(stock_entry, method):
 			comment.save()
 
 	frappe.db.commit()
+
+def raw_material_update_on_bom():
+	past_seven_days = get_date_str(add_days(today(), -7))
+	todays_date = today()
+	boms = frappe.get_all("BOM", filters= {
+		"manufacturing_type" : "Process"
+	},
+	fields=["name"]
+	)
+	for bom in boms:
+		stocks = frappe.get_all("Stock Entry", filters={
+			"bom_no": bom.name,
+			"posting_date": ["BETWEEN", [past_seven_days, todays_date]]
+		})
+		raw_material = 0
+		fg= 0
+		avg_manufactured_qty = 0
+		for stock in stocks:
+			stock_entry = frappe.get_doc("Stock Entry", {"name": stock.name, "bom_no": bom.name})
+			for item in stock_entry.items:
+				if item.s_warehouse:
+					raw_material = raw_material + item.qty
+				elif item.t_warehouse:
+					fg = fg + item.qty
+		if fg and raw_material:
+			avg_manufactured_qty= fg/raw_material
+		frappe.db.set_value("BOM", bom.name, "avg_manufactured_qty", avg_manufactured_qty)
```

Comment:
Move this ERPNext

Reply:
Okay

Reply:
I meant add all the contents of PR in ERPNext only